packages/@azure/servicebus/data-plane/test/streamingReceiver.spec.ts

```diff
@@ -300,89 +301,162 @@ describe("Complete message", function(): void {
+  });
+
+  async function testMultipleAbandons(): Promise<void> {
+    await senderClient.sendBatch(testSimpleMessages);
+
+    let checkDeliveryCount0 = 0;
+    let checkDeliveryCount1 = 0;
+
+    const receiveListener = await receiverClient.receive(
+      (msg: ServiceBusMessage) => {
+        console.log(checkDeliveryCount0);
+        if (msg.messageId === testSimpleMessages[0].messageId) {
+          should.equal(msg.deliveryCount, checkDeliveryCount0, "Unexpected deliveryCount.");
+          checkDeliveryCount0++;
+        } else if (msg.messageId === testSimpleMessages[1].messageId) {
+          should.equal(msg.deliveryCount, checkDeliveryCount1, "Unexpected deliveryCount.");
+          checkDeliveryCount1++;
+        }
+        return msg.abandon();
+      },
+      unExpectedErrorHandler,
+      { autoComplete: false }
+    );
+
+    await delay(6000);
+
+    await receiveListener.stop();
+    should.equal(unexpectedError, undefined, unexpectedError && unexpectedError.message);
+
+    should.equal(checkDeliveryCount0, maxDeliveryCount);
+    should.equal(checkDeliveryCount1, maxDeliveryCount);
+
+    await testPeekMsgsLength(receiverClient, 0); // No messages in the queue
+
+    const deadLetterMsgs = await deadLetterClient.receiveBatch(2);
+    should.equal(Array.isArray(deadLetterMsgs), true);
+    should.equal(deadLetterMsgs.length, testSimpleMessages.length);
+    should.equal(deadLetterMsgs[0].deliveryCount, maxDeliveryCount);
+    should.equal(deadLetterMsgs[1].deliveryCount, maxDeliveryCount);
+    should.equal(testSimpleMessages.some((x) => deadLetterMsgs[0].messageId === x.messageId), true);
+    should.equal(testSimpleMessages.some((x) => deadLetterMsgs[1].messageId === x.messageId), true);
+
+    await deadLetterMsgs[0].complete();
+    await deadLetterMsgs[1].complete();
+
+    await testPeekMsgsLength(deadLetterClient, 0);
+  }
+
+  it("Partitioned Queue: Multiple abandons until maxDeliveryCount.", async function(): Promise<
+    void
+  > {
+    await beforeEachTest(ClientType.PartitionedQueue, ClientType.PartitionedQueue);
+    await testMultipleAbandons();
+  });
+
+  it("Partitioned Topics and Subscription: Multiple abandons until maxDeliveryCount.", async function(): Promise<
+    void
+  > {
+    await beforeEachTest(ClientType.PartitionedTopic, ClientType.PartitionedSubscription);
+    await testMultipleAbandons();
+  });
+
+  it("Unpartitioned Queue: Multiple abandons until maxDeliveryCount.", async function(): Promise<
+    void
+  > {
+    await beforeEachTest(ClientType.UnpartitionedQueue, ClientType.UnpartitionedQueue);
+    await testMultipleAbandons();
+  });
+
+  it("Unpartitioned Topics and Subscription: Multiple abandons until maxDeliveryCount.", async function(): Promise<
+    void
+  > {
+    await beforeEachTest(ClientType.UnpartitionedTopic, ClientType.UnpartitionedSubscription);
+    await testMultipleAbandons();
+  });
+  //   async function testAbandon(autoComplete: boolean): Promise<void> {
```

Comment:
We can remove these. Any reason you want to keep them?

Reply:
Keep them in comments till we fix them ?

Reply:
The only case they were testing was that when message is abandoned, it goes back to the queue with incremented delivery count.
The same is also essentially done in the multiple abandon case, so we can remove this section.
For that matter, we can also remove the `testAbandon` for the batchingReceiver and just keep the multiple abandon test case

Reply:
Same with batchingReceiver ?

Reply:
Yes

Reply:
But for now, lets not add more work for ourselves, we have a few bugs to look at. I am ok keeping both `testAbandon` and `testMultipleAbandon` for batching, but only have `testMutlipleAbaddon` for streaming

Reply:
Okay, batching has both and streaming has only `testMultipleAbandon`.